src/ImageSharp/Processing/Transforms/Transform.cs

```diff
@@ -0,0 +1,116 @@
+            where TPixel : struct, IPixel<TPixel>
+            => Transform(source, matrix, sampler, Size.Empty);
+
+        /// <summary>
+        /// Transforms an image by the given matrix using the specified sampling algorithm.
+        /// </summary>
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
+        /// <param name="source">The image to transform.</param>
+        /// <param name="matrix">The transformation matrix.</param>
+        /// <param name="sampler">The <see cref="IResampler"/> to perform the resampling.</param>
+        /// <param name="sourceRectangle">The rectangle defining the source pixel area to transform. 'sourceRectangle.Location' becomes the origo of the transformed image.</param>
+        /// <returns>The <see cref="Image{TPixel}"/></returns>
+        public static IImageProcessingContext<TPixel> Transform<TPixel>(
+            this IImageProcessingContext<TPixel> source,
+            Matrix3x2 matrix,
+            IResampler sampler,
+            Rectangle sourceRectangle)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            var t = Matrix3x2.CreateTranslation(-sourceRectangle.Location);
+            Matrix3x2 combinedMatrix = t * matrix;
+            return source.ApplyProcessor(new AffineTransformProcessor<TPixel>(combinedMatrix, sampler, sourceRectangle.Size));
+        }
+
+        /// <summary>
+        /// Transforms an image by the given matrix using the specified sampling algorithm.
+        /// </summary>
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
+        /// <param name="source">The image to transform.</param>
+        /// <param name="matrix">The transformation matrix.</param>
+        /// <param name="sampler">The <see cref="IResampler"/> to perform the resampling.</param>
+        /// <param name="destinationSize">The dimensions to constrain the transformed image to.</param>
+        /// <returns>The <see cref="Image{TPixel}"/></returns>
+        public static IImageProcessingContext<TPixel> Transform<TPixel>(
+            this IImageProcessingContext<TPixel> source,
+            Matrix3x2 matrix,
+            IResampler sampler,
+            Size destinationSize)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            return source.ApplyProcessor(new AffineTransformProcessor<TPixel>(matrix, sampler, destinationSize));
+        }
+
+        /// <summary>
+        /// Transforms an image by the given matrix.
+        /// </summary>
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
+        /// <param name="source">The image to transform.</param>
+        /// <param name="matrix">The transformation matrix.</param>
+        /// <returns>The <see cref="Image{TPixel}"/></returns>
+        internal static IImageProcessingContext<TPixel> Transform<TPixel>(this IImageProcessingContext<TPixel> source, Matrix4x4 matrix)
```

Comment:
Good that it's internal for now. How about finishing projective transforms in a follow-up PR? I'd rather have this merged first, it's already *The PR That Has Been Under Review For The Longest Time In History* :)

Reply:
Absolutely! Merge this as soon as you are satisfied with it.